### PR TITLE
ENH: GH-35611 Tests for top-level Pandas functions serializable

### LINKED ISSUE #35611: ENH: Make top-level Pandas functions serializable
```python
import pandas as pd
import pickle
pickle.dumps(pd.read_csv)
```

```python-traceback
AttributeError: Can't pickle local object '_make_parser_function.<locals>.parser_f'
```

### PR DESCRIPTION
**Pull request by ikedaosushi**
- [x] closes #35611
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/test_common.py [hunk 1]
```diff
@@ -157,3 +158,10 @@ def test_version_tag():
         raise ValueError(
             "No git tags exist, please sync tags between upstream and your repo"
         )
+
+
+def test_serializable():
```
Comment (jreback) [inline pandas/tests/test_common.py:163]: can you parameterize the test on these instead or inside

also pls add the issue number here as a comment

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_common.py (hunk 1 above)
Comment (ikedaosushi) [reply pandas/tests/test_common.py:163]: I understand. I'll do that.

Comment (ikedaosushi) [reply pandas/tests/test_common.py]: @jreback Could you take a look of this?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ikedaosushi

**Pull request merged**